name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TastytradePDFExtractor.java

```diff
@@ -0,0 +1,529 @@
+     * It summarizes the trades of one day and is issued as soon as the trades are settled, i.e. T+2.
+     * The Trade Confirmation document only contains trades, i.e. no general account transactions. 
+     */
+    private void addSummaryStatementBuySellTransaction()
+    {
+        // The document does not contain any sort of identifation markings such as "Trade Confirmation"
+        // However, the name TASTYWORKS, INC appears on the first page.
+        DocumentType type = new DocumentType("TASTYWORKS, INC");
+        this.addDocumentTyp(type);
+
+        // A block is a stock trade, if it contains a ticker symbol after the settlement date.
+        // 2 B 07/17/23 07/19/23 8 VUG 290.5298000 2,324.24 0.00 0.00 0.00 D2851 2,324.24 TUA0719 6 1
+        Block stockTrade = new Block("^[\\d][\\s]+.[\\s]+[\\d]{2}/[\\d]{2}/[\\d]{2}[\\s]+[\\d]{2}/[\\d]{2}/[\\d]{2}[\\s]+[,\\d]+[\\s]+[\\w]+[\\s]+[\\.,\\d]+.*");
+        type.addBlock(stockTrade);
+        stockTrade.set(new Transaction<BuySellEntry>().subject(() -> {
+            BuySellEntry entry = new BuySellEntry();
+            entry.setType(PortfolioTransaction.Type.BUY);
+            entry.setCurrencyCode("USD");
+            return entry;
+        })
+            .section("addlfee", "commfee", "currency", "cusip", "name", "amount", "shares",
+                            "tickerSymbol", "tradedate", "tranfee", "type").optional()
+            .match("^[\\d][\\s]+(?<type>.)"
+                            + "[\\s]+(?<tradedate>[\\d]{2}/[\\d]{2}/[\\d]{2})[\\s]+[\\d]{2}/[\\d]{2}/[\\d]{2}"
+                            + "[\\s]+(?<shares>[,\\d]+)"
+                            + "[\\s]+(?<tickerSymbol>[\\w]+)[\\s]+[\\.,\\d]+[\\s]+[\\.,\\d]+"
+                            + "[\\s]+(?<commfee>[\\.,\\d]+)"
+                            + "[\\s]+(?<tranfee>[\\.,\\d]+)"
+                            + "[\\s]+(?<addlfee>[\\.,\\d]+)[\\s]+[\\w]+"
+                            + "[\\s]+(?<amount>[\\.,\\d]+)[\\s]+[\\w\\d]+ [\\d] [\\d] *$")
+            // Desc: VANGUARD INDEX FUNDS VANGUARD GROWTH ETF Interest/STTax: 0.00 CUSIP: 922908736
+            .match("^Desc:[\\s]+(?<name>.+)[\\s]+Interest/STTax:[\\s]+[\\.,\\d]+[\\s]+CUSIP:[\\s]+(?<cusip>[\\w]+)$")
+            .match("^Currency:[\\s]+(?<currency>[\\w]{3})[\\s]+ReportedPX: .*$")
+            .assign((t, v) -> {
+                if (v.get("type").equals("S"))
+                    t.setType(PortfolioTransaction.Type.SELL);
+                v.put("wkn", v.get("cusip"));
+                t.setSecurity(getOrCreateSecurity(v));
+                t.setDate(asDate(v.get("tradedate")));
+                t.setShares(asShares(v.get("shares"), "en", "US"));
+                t.setAmount(asAmount(v.get("amount"), "en", "US"));
+                Money fee = Money.of("USD", asAmount(v.get("commfee"), "en", "US")
+                                + asAmount(v.get("tranfee"), "en", "US")
+                                + asAmount(v.get("addlfee"), "en", "US"));
```

Comment:
Would be more helpful to calculate an amount via the [Money.class](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/money/Money.java).
This way we have the check that the currencies are also correct to each other.

Reply:
The currency is always USD. Internally, Money uses a `long` member to represent the total amount. The method `asAmount(..)` will return a `long`. That means using the Money class should not make any difference, computation-wise. But it would be more clear and readable though. The current version is just shorter.
We will do whatever you prefer.